ur_robot_driver/include/ur_robot_driver/ros/data_field_publisher.h

```diff
@@ -0,0 +1,176 @@
+// this is for emacs file handling -*- mode: c++; indent-tabs-mode: nil -*-
+
+// -- BEGIN LICENSE BLOCK ----------------------------------------------
+// Copyright 2019 FZI Forschungszentrum Informatik
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -- END LICENSE BLOCK ------------------------------------------------
+
+//----------------------------------------------------------------------
+/*!\file
+ *
+ * \author  Tristan Schnell schnell@fzi.de
+ * \date    2019-10-30
+ *
+ */
+//----------------------------------------------------------------------
+#ifndef UR_DRIVER_DATA_FIELD_PUBLISHER_H_INCLUDED
+#define UR_DRIVER_DATA_FIELD_PUBLISHER_H_INCLUDED
+
+#include <realtime_tools/realtime_publisher.h>
+#include <ur_robot_driver/rtde/data_package.h>
+
+namespace ur_driver
+{
+namespace rtde_interface
+{
+/*!
+ * \brief The DataFieldPublisher class implements an abstract wrapper for various ROS publishers
+ * that publish fields of the RTDE data package. In addition, it contains a static factory to
+ * create correct publishers for a data field.
+ */
+class DataFieldPublisher
+{
+public:
+  DataFieldPublisher() = default;
+
+  /*!
+   * \brief Publishes partial information from a data package.
+   *
+   * \param data_package The given data package to publish from
+   *
+   * \returns True if the realtime publisher could publish the data.
+   */
+  virtual bool publish(const std::unique_ptr<DataPackage>& data_package) = 0;
```

Comment:
Would passing a plain `DataPackage const&` work here? If there is no strong need for `DataFieldPublisher`s to have access to the pointer, a plain reference would seem better.